### ISSUE TITLE
[DataGrid] Keyboard navigation issue with pagination

### ISSUE THREAD

**Issue by cherniavskii** [mui/mui-x#5339]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

After switching the page with a keyboard there's no way to move focus back to grid cells with a keyboard (`Shift+Tab`)

### Expected behavior 🤔

After switching the page with a keyboard you should be able to move focus back to grid cells

### Steps to reproduce 🕹

Steps:

1. Go to https://codesandbox.io/s/pagesizecustomoptions-demo-mui-x-forked-fdvvoc
2. Focus any cell
3. Press `Tab` twice to focus Next page button
4. Press `Enter` to switch to the next page
5. Press `Shift+Tab` 3 times

See that you cannot focus grid cell with a keyboard anymore.

Note, that if you skip step 4, it works just fine

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by gd185126** [comment]
@cherniavskii Has there been any updates on this ticket?

**Comment by gd185126** [comment]
@cherniavskii @alexfauquette Are there any updates on the status of this ticket?

**Comment by alexfauquette** [comment]
A solution could be to verify that the tabbable cell is included in the current page. Otherwise, it's the first column header that takes `tabIndex=0`.

https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx#L287-L292

**Comment by ac185422** [comment]
Is there a way to do this currently though I'm not sure what that link to hook is for @alexfauquette. I know for example the onCellFocusOut callback defined here: https://mui.com/x/api/data-grid/data-grid/ is fired if we're focused on a cell, and then go to next page, is there a way in this callback that we can be setting the tabindex or resetting focus to initial state as if the table was just loaded so it is back to the first column header when we tab into it for example?
